systemtest/src/test/java/io/strimzi/systemtest/security/NetworkPoliciesST.java

```diff
@@ -0,0 +1,334 @@
+
+        checkNetworkPoliciesInNamespace(NAMESPACE);
+
+        changeKafkaConfigurationAndCheckObservedGeneration(NAMESPACE);
+    }
+
+    @Test
+    void testNPWhenOperatorIsInDifferentNamespaceThanOperand() {
+        String secondNamespace = "second-" + NAMESPACE;
+
+        Map<String, String> labels = new HashMap<>();
+        labels.put("my-label", "my-value");
+
+        EnvVar operatorLabelsEnv = new EnvVarBuilder()
+            .withName("STRIMZI_OPERATOR_NAMESPACE_LABELS")
+            .withValue(labels.toString().replaceAll("\\{|}", ""))
+            .build();
+
+        cluster.createNamespace(secondNamespace);
+
+        prepareEnvForOperator(NAMESPACE, Arrays.asList(NAMESPACE, secondNamespace));
+
+        // Apply role bindings in CO namespace
+        applyRoleBindings(NAMESPACE);
+
+        // Create ClusterRoleBindings that grant cluster-wide access to all OpenShift projects
+        List<ClusterRoleBinding> clusterRoleBindingList = KubernetesResource.clusterRoleBindingsForAllNamespaces(NAMESPACE);
+        clusterRoleBindingList.forEach(clusterRoleBinding ->
+            KubernetesResource.clusterRoleBinding(clusterRoleBinding, NAMESPACE));
+        // 060-Deployment
+        BundleResource.clusterOperator("*", Constants.CO_OPERATION_TIMEOUT_DEFAULT)
+            .editOrNewSpec()
+                .editOrNewTemplate()
+                    .editOrNewSpec()
+                        .editContainer(0)
+                            .addToEnv(operatorLabelsEnv)
+                        .endContainer()
+                    .endSpec()
+                .endTemplate()
+            .endSpec()
+            .done();
+
+        kubeClient().getClient().namespaces().withName(NAMESPACE).edit()
+            .editMetadata()
+                .addToLabels(labels)
+            .endMetadata()
+            .done();
+
+        cluster.setNamespace(secondNamespace);
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3)
+            .editMetadata()
+                .addToLabels(labels)
+            .endMetadata()
+            .done();
+
+        checkNetworkPoliciesInNamespace(secondNamespace);
+
+        changeKafkaConfigurationAndCheckObservedGeneration(secondNamespace);
+    }
+
+    void checkNetworkPoliciesInNamespace(String namespace) {
+        List<NetworkPolicy> networkPolicyList = new ArrayList<>(kubeClient().getClient().network().networkPolicies().inNamespace(namespace).list().getItems());
+
+        String networkPolicyPrefix = CLUSTER_NAME + "-network-policy";
+
+        assertNotNull(networkPolicyList.stream().filter(networkPolicy ->  networkPolicy.getMetadata().getName().contains(networkPolicyPrefix + "-kafka")).findFirst());
+        assertNotNull(networkPolicyList.stream().filter(networkPolicy ->  networkPolicy.getMetadata().getName().contains(networkPolicyPrefix + "-zookeeper")).findFirst());
+    }
+
+    void changeKafkaConfigurationAndCheckObservedGeneration(String namespace) {
```

Comment:
Why this check is needed

Reply:
When you don't have created NP for Kafka, you cannot do requests on Kafka API. If I'm correct, the dynamic configuration is changed through the Kafka API, so these checks will ensure us the config is applied and NP are working. (the observed generation is higher after these changes)

Reply:
Ok thanks :)